src/darwin/mod.rs

```diff
@@ -50,21 +45,27 @@ struct Descriptor {
     style_name: String,
     font_path: PathBuf,
 
-    ct_descriptor: CTFontDescriptor,
+    ct_descriptor: CFRetained<CTFontDescriptor>,
 }
 
 impl Descriptor {
-    fn new(desc: CTFontDescriptor) -> Descriptor {
-        Descriptor {
-            style_name: desc.style_name(),
-            font_path: desc.font_path().unwrap_or_default(),
-            ct_descriptor: desc,
-        }
+    fn new(desc: &CTFontDescriptor) -> Descriptor {
+        let style_name = unsafe { desc.attribute(kCTFontStyleNameAttribute) }
```

Comment:
This is a rather unusual unsafe usage, I think I'd prefer it being assigned to its own variable over chaining off its scope block.

Also what exactly makes this unsafe? I'd assume it's due to the parameter potentially not existing? Might be a good idea to have a safety comment on it.